q2_moshpit/busco/utils.py

```diff
@@ -0,0 +1,384 @@
+import os
+import pandas as pd
+import altair as alt
+import seaborn as sns
+import matplotlib.pyplot as plt
+from zipfile import ZipFile
+from .._utils import run_command
+from copy import deepcopy
+from typing import List, Dict
+from q2_types_genomics.per_sample_data._format import MultiMAGSequencesDirFmt
+
+arguments_with_hyphens = {
+    "auto_lineage": "auto-lineage",
+    "auto_lineage_euk": "auto-lineage-euk",
+    "auto_lineage_prok": "auto-lineage-prok",
+    "list_datasets": "list-datasets",
+    "update_data": "update-data",
+}
+
+
+def _parse_busco_params(arg_key, arg_val) -> List[str]:
+    """Creates a list with argument and its value to be consumed by MetaBAT 2.
+
+    Argument names will be converted to command line parameters by
+    appending a '--' prefix and concatenating words separated by a '_',
+    e.g.: 'some_parameter_x' -> '--someParameterX'.
+
+    Args:
+        arg_key (str): Argument name.
+        arg_val: Argument value.
+
+    Returns:
+        [converted_arg, arg_value]: List containing a prepared command line
+            parameter and, optionally, its value.
+    """
+
+    # If the key is one of
+    if arg_key in arguments_with_hyphens.keys():
+        arg_key = arguments_with_hyphens[arg_key]
```

Comment:
The problem here is that some of the arguments that must be passed to busco have `-` while others have `_`, such that one cannot simply apply one rule for all arguments. Since arguments in Python cannot have `-` this means transforming all busco `-` parameters from `_` to `-` which is what I try to do in this function.

Reply:
There will never be an argument with hyphens since all keyword arguments that are registered in the main function only contain underscores. As for the second requested test ("one for an arg which you don't want to rename"), isn't the [`test_parse_busco_params_2`](https://github.com/Sann5/q2-moshpit/blob/173415bf90cad7edba79e6256aaf8199f1ea2f89/q2_moshpit/busco/tests/test_busco.py#L49C22-L49C22) already testing this?